src/qml/qmlapplication.cpp

```diff
@@ -5,6 +5,7 @@
 #include "control/controlsortfiltermodel.h"
 #include "controllers/controllermanager.h"
 #include "moc_qmlapplication.cpp"
+#include "preferences/dialog/dlgpreferences.h"
```

Comment:
This line seems to cause issue since it means that the QML library (in my feature, I have splitted `mixxx-lib` in two) will also need `preferences/dialog/ui_dlgpreferencesdlg.h`. I belive however that I could take your change in, plus the one I've made to initialize the QML signletons as part of the CoreService instead of as part of the QML application, which then removes the need for `DlgPreferences` in this file

Reply:
I believe this line can actually be removed.